interface/wx/httprequest.h

```diff
@@ -0,0 +1,232 @@
+
+    @since 3.1.2
+    @library{wxnet}
+    @category{net}
+
+    @see wxHTTPSession
+*/
+class wxHTTPRequest
+{
+public:
+    /**
+        Create a new request instance
+
+        @param url The URL of the HTTP resource for this request
+        @param session If no wxHTTPSession is give the default session will
+                       be used
+
+        @since 3.1.2
+    */
+    wxHTTPRequest(const wxString& url, wxHTTPSession session = NULL);
+
+    /**
+
+        @since 3.1.2
+    */
+    ~wxHTTPRequest();
+
+    /**
+        Adds a request header send by this request.
+
+        @param name Name if the header
+        @param value String value of the header
+
+        @since 3.1.2
+    */
+    void AddRequestHeader(const wxString& name, const wxString& value);
+
+    /**
+        Set HTTP method.
+
+        Set <a href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html">common</a>
+        or expanded HTTP method.
+
+        Overrides GET or POST methods that is used by default.
+
+        @param method
+            HTTP method name, e.g. "GET".
+
+        @since 3.1.2
+    */
+    void SetRequestMethod(const wxString& method);
+
+    /**
+        Set the text to be posted to the server.
+
+        After a successful call to this method, the request will use HTTP @c
+        POST instead of the default @c GET when it's executed.
+
+        @param data
+            The data to post.
+        @param contentType
+            The value of HTTP "Content-Type" header, e.g. "text/html;
+            charset=UTF-8".
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxString& data, const wxString& contentType);
+
+    /**
+        Set the binary data to be posted to the server.
+
+        The next request will
+        be an HTTP @c POST instead of the default HTTP @c GET and the given @a
+        data will be posted as the body of this request.
+
+        @since 3.1.2
+    */
+    void SetRequestData(const wxInputStream*, const wxString& contentType);
+
+    /**
+        Send the request to the server
+
+        @return
+            @true if no error occured or @false if a client or server side
+            error was triggered
+
+        @since 3.1.2
+    */
+    bool Send();
+
+    /**
+        Return a header from the response
+
+        @param name Name of the header field
+
+        @since 3.1.2
+    */
+    wxString GetResponseHeader(const wxString& name) const;
+
+    /**
+        Returns the status code returned by the server
+
+        @since 3.1.2
+    */
+    int GetResponseStatus() const;
+
+    /**
+        Returns the response status text
+
+        @since 3.1.2
+    */
+    wxString GetResponseStatusText() const;
+
+    /**
+        Returns a stream which represents the reponse data sent by the server.
+
+        @since 3.1.2
+    */
+    wxInputStream* GetResponseStream();
+
+    /**
+        Returns all response data as a string.
+
+        @since 3.1.2
+    */
+    wxString GetResponseData();
```

Comment:
Do we really want to return this by value?

Also, I'm pretty sure we do _not_ want to return it as `wxString`, this won't work well for binary/non-UTF-8 data. We could have `GetResponseText(wxMBConv conv = wxConvUTF8)` for convenience, but I think we also need a lower level method providing direct access to the underlying `wxMemoryBuffer` or maybe `wxVector<wxUint8>` or something like that.

Reply:
Wasn't meant for binary data. So `GetResponseText()` is more appropriate. I would use conv as a pointer, defaulting to NULL which would determine based on the response if possible or defaulting to UTF-8